### PR TITLE
Fix jaeger exporter by assigning jaegerAgentUdpBatcher in constructor

### PR DESCRIPTION
**Pull request by cijothomas**
Also modified sample and readme to reflect recent changes

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Jaeger/JaegerTraceExporter.cs [hunk 1]
```diff
@@ -34,14 +34,10 @@ public class JaegerTraceExporter : SpanExporter, IDisposable
         private readonly IJaegerUdpBatcher jaegerAgentUdpBatcher;
         private bool disposedValue = false; // To detect redundant dispose calls
 
-        public JaegerTraceExporter(JaegerExporterOptions options)
+        public JaegerTraceExporter(JaegerExporterOptions options, IJaegerUdpBatcher jaegerAgentUdpBatcher)
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Jaeger/JaegerTraceExporter.cs]: Alternatively we can keep the existing constructor and default to  use `JaegerUdpBatcher`

Comment (lmolkova) [reply src/OpenTelemetry.Exporter.Jaeger/JaegerTraceExporter.cs]: Can you initialize batcher in this constructor and add another constructor that accepts batcher?

File: src/OpenTelemetry.Exporter.Jaeger/JaegerTraceExporter.cs [hunk 2]
```diff
@@ -34,6 +34,13 @@ public class JaegerTraceExporter : SpanExporter, IDisposable
         private readonly IJaegerUdpBatcher jaegerAgentUdpBatcher;
         private bool disposedValue = false; // To detect redundant dispose calls
 
+        public JaegerTraceExporter(JaegerExporterOptions options)
```
Comment (lmolkova) [inline src/OpenTelemetry.Exporter.Jaeger/JaegerTraceExporter.cs]: can you do `this(options, new JaegerUdpBatcher(options))` so we'll not need to change both ctors if we decide to change something?

**Comment by cijothomas** [comment]
#please-merge

**Comment by cijothomas** [comment]
@lmolkova can you merge this?

**Pull request merged**